provider/lxd/environ_broker_test.go

```diff
@@ -406,6 +406,50 @@ func (s *environBrokerSuite) TestStartInstanceWithConstraints(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *environBrokerSuite) TestStartInstanceWithConstraintsAndVirtType(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	svr := lxd.NewMockServer(ctrl)
+
+	// Check that the constraints were passed through to spec.Config.
+	check := func(spec containerlxd.ContainerSpec) bool {
+		cfg := spec.Config
+		if cfg["limits.cpu"] != "2" {
+			return false
+		}
+		if cfg["limits.memory"] != "2048MiB" {
+			return false
+		}
+		return spec.InstanceType == "t2.micro" && spec.VirtType == api.InstanceTypeVM
+	}
+
+	exp := svr.EXPECT()
+	gomock.InOrder(
```

Comment:
It looks like InOrder is not strictly necessary here. We're trying to move away from using it where possible.